custom_components/kia_uvo/__init__.py

```diff
@@ -235,6 +229,9 @@ async def async_update_options(hass: HomeAssistant, config_entry: ConfigEntry):
 
 
 async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry):
+    current_data = config_entry.data.copy()
+    current_data[CONF_STORED_CREDENTIALS] = vars(vehicle.token)
+    hass.config_entries.async_update_entry(config_entry, data=current_data)
```

Comment:
do we also need to supply `options` argument too, so they are not lost between saves? not sure what is happening when `options` is missing